enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -4919,6 +4929,416 @@ class AdjointGenerator
         }
       }
 
+      if (Mode == DerivativeMode::ReverseModeGradient) {
+        eraseIfUnused(call, /*erase*/ true, /*check*/ false);
+      } else {
+        eraseIfUnused(call);
+      }
+    }
+
+    if ((funcName == "cblas_sswap" || funcName == "cblas_dswap") &&
+        called->isDeclaration()) {
+      // sswap(int n, float *x, int incx, float *y, int incy)
```

Comment:
You may also need to cache n, incx, incy here if they are passed by reference as demonstrated in https://github.com/EnzymeAD/Enzyme/pull/528

Reply:
I think I would first merge the code I currently have, which haven't handled pass-by-reference. This is already useful in many cases (c/c++ applications and Enzyme.jl). Handling fortran-style BLAS is way more complicated so that it would be better done when we have TableGen